#### 1.1 Utilize a pretrained YOLO model
we've chose yolov8

In [ ]:
%pip install roboflow

     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 74.6/74.6 kB 1.6 MB/s eta 0:00:00
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 158.3/158.3 kB 6.4 MB/s eta 0:00:00
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 178.7/178.7 kB 11.7 MB/s eta 0:00:00
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 58.8/58.8 kB 6.7 MB/s eta 0:00:00
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 49.1/49.1 MB 14.5 MB/s eta 0:00:00
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 54.5/54.5 kB 5.1 MB/s eta 0:00:00
  Attempting uninstall: opencv-python-headless
    Found existing installation: opencv-python-headless 4.9.0.80
    Uninstalling opencv-python-headless-4.9.0.80:
      Successfully uninstalled opencv-python-headless-4.9.0.80
  Attempting uninstall: idna
    Found existing installation: idna 3.7
    Uninstalling idna-3.7:
      Successfully uninstalled idna-3.7
  Attempting uninstall: cycler
    Found existing installation: cycler 0.12.1
    Uninstalling cycler-0.12.1:
      Successfully uninstalled cycler-0.1

In [ ]:
# Install YOLOv8
%pip install ultralytics

     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 756.9/756.9 kB 7.9 MB/s eta 0:00:00
  Using cached nvidia_cuda_nvrtc_cu12-12.1.105-py3-none-manylinux1_x86_64.whl (23.7 MB)
  Using cached nvidia_cuda_runtime_cu12-12.1.105-py3-none-manylinux1_x86_64.whl (823 kB)
  Using cached nvidia_cuda_cupti_cu12-12.1.105-py3-none-manylinux1_x86_64.whl (14.1 MB)
  Using cached nvidia_cudnn_cu12-8.9.2.26-py3-none-manylinux1_x86_64.whl (731.7 MB)
  Using cached nvidia_cublas_cu12-12.1.3.1-py3-none-manylinux1_x86_64.whl (410.6 MB)
  Using cached nvidia_cufft_cu12-11.0.2.54-py3-none-manylinux1_x86_64.whl (121.6 MB)
  Using cached nvidia_curand_cu12-10.3.2.106-py3-none-manylinux1_x86_64.whl (56.5 MB)
  Using cached nvidia_cusolver_cu12-11.4.5.107-py3-none-manylinux1_x86_64.whl (124.2 MB)
  Using cached nvidia_cusparse_cu12-12.1.0.106-py3-none-manylinux1_x86_64.whl (196.0 MB)
  Using cached nvidia_nccl_cu12-2.19.3-py3-none-manylinux1_x86_64.whl (166.0 MB)
  Using cached nvidia_nvtx_cu12-12.1.105-py3-none-many

### download dataset

In [1]:
from roboflow import Roboflow
rf = Roboflow(api_key="lBm4iE4T4ZnkvuavmAWR")
project = rf.workspace("roboflow-universe-projects").project("soccer-players-ckbru")
version = project.version(16)
dataset = version.download("yolov8")

loading Roboflow workspace...
loading Roboflow project...
Dependency ultralytics==8.0.196 is required but found version=8.2.15, to fix: `pip install ultralytics==8.0.196`



Extracting Dataset Version Zip to Soccer-Players-16 in yolov8:: 100%|██████████| 312/312 [00:00<00:00, 608.33it/s]


### Training the model

In [ ]:
# Training
!yolo detect train model=yolov8n.pt \
    data=/content/Soccer-Players-16/data.yaml\
    workers=2 imgsz=640 batch=64 device=0 epochs=100 patience=50 name=yolov8_Scoccer_Players_default

Ultralytics YOLOv8.2.15 🚀 Python-3.10.12 torch-2.2.1+cu121 CUDA:0 (Tesla T4, 15102MiB)
engine/trainer: task=detect, mode=train, model=yolov8n.pt, data=/content/Soccer-Players-16/data.yaml, epochs=100, time=None, patience=50, batch=64, imgsz=640, save=True, save_period=-1, cache=False, device=0, workers=2, project=None, name=yolov8_Scoccer_Players_default2, exist_ok=False, pretrained=True, optimizer=auto, verbose=True, seed=0, deterministic=True, single_cls=False, rect=False, cos_lr=False, close_mosaic=10, resume=False, amp=True, fraction=1.0, profile=False, freeze=None, multi_scale=False, overlap_mask=True, mask_ratio=4, dropout=0.0, val=True, split=val, save_json=False, save_hybrid=False, conf=None, iou=0.7, max_det=300, half=False, dnn=False, plots=True, source=None, vid_stride=1, stream_buffer=False, visualize=False, augment=False, agnostic_nms=False, classes=None, retina_masks=False, embed=None, show=False, save_frames=False, save_txt=False, save_conf=False, save_crop=False, show_l

### Evaluation
                 Class     Images  Instances      Box(P          R      mAP50  m        
                   all         28        401      0.954      0.782      0.831      0.516
                  Ball         28         23          1      0.421      0.522      0.261
                Player         28        346      0.916      0.988      0.993      0.681
                   Ref         28         32      0.946      0.938      0.978      0.605

In [5]:
# Training
!yolo detect val model=old_best.pt \
    data=data.yaml\
    workers=2 imgsz=640 batch=64 device=0 epochs=10

Ultralytics YOLOv8.2.15 🚀 Python-3.11.9 torch-2.3.0 CUDA:0 (NVIDIA GeForce GTX 1650, 4096MiB)
Model summary (fused): 168 layers, 3006233 parameters, 0 gradients, 8.1 GFLOPs
val: New cache created: C:\Users\Mo\PycharmProjects\yolo-project\datasets\Soccer-Players-16\valid\labels.cache
                   all         28        401      0.937      0.762      0.813      0.522
                  Ball         28         23      0.963      0.391      0.494      0.272
                Player         28        346      0.912      0.983      0.992      0.695
                   Ref         28         32      0.936      0.912      0.954      0.598
Speed: 2.3ms preprocess, 15.3ms inference, 0.0ms loss, 100.5ms postprocess per image
Results saved to runs\detect\val5
💡 Learn more at https://docs.ultralytics.com/modes/val



val: Scanning C:\Users\Mo\PycharmProjects\yolo-project\datasets\Soccer-Players-16\valid\labels...:   0%|          | 0/28 [00:00<?, ?it/s]
val: Scanning C:\Users\Mo\PycharmProjects\yolo-project\datasets\Soccer-Players-16\valid\labels... 2 images, 0 backgrounds, 0 corrupt:   7%|7         | 2/28 [00:00<00:01, 19.22it/s]
val: Scanning C:\Users\Mo\PycharmProjects\yolo-project\datasets\Soccer-Players-16\valid\labels... 28 images, 1 backgrounds, 0 corrupt: 100%|##########| 28/28 [00:00<00:00, 177.34it/s]

                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95):   0%|          | 0/1 [00:00<?, ?it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|##########| 1/1 [00:16<00:00, 16.91s/it]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|##########| 1/1 [00:16<00:00, 16.91s/it]


### hyperparameter tuning

In [ ]:
# Training with Tuned Hyperparameters
!yolo detect train model=/content/runs/detect/yolov8_Scoccer_Players_default2/weights/best.pt data=/content/Soccer-Players-16/data.yaml name=yolov8_Scoccer_Players_tuned device=0 epochs=100 patience=50 imgsz=640 workers=2 batch=64 lr0=0.01 lrf=0.01 momentum=0.95 weight_decay=0.0001 warmup_epochs=10 warmup_momentum=0.5 warmup_bias_lr=0.1 optimizer=SGD

Ultralytics YOLOv8.2.15 🚀 Python-3.10.12 torch-2.2.1+cu121 CUDA:0 (Tesla T4, 15102MiB)
engine/trainer: task=detect, mode=train, model=/content/runs/detect/yolov8_Scoccer_Players_default2/weights/best.pt, data=/content/Soccer-Players-16/data.yaml, epochs=100, time=None, patience=50, batch=64, imgsz=640, save=True, save_period=-1, cache=False, device=0, workers=2, project=None, name=yolov8_Scoccer_Players_tuned6, exist_ok=False, pretrained=True, optimizer=SGD, verbose=True, seed=0, deterministic=True, single_cls=False, rect=False, cos_lr=False, close_mosaic=10, resume=False, amp=True, fraction=1.0, profile=False, freeze=None, multi_scale=False, overlap_mask=True, mask_ratio=4, dropout=0.0, val=True, split=val, save_json=False, save_hybrid=False, conf=None, iou=0.7, max_det=300, half=False, dnn=False, plots=True, source=None, vid_stride=1, stream_buffer=False, visualize=False, augment=False, agnostic_nms=False, classes=None, retina_masks=False, embed=None, show=False, save_frames=False, s

### evaluation

In [ ]:
!yolo detect val model=new_best.pt \
    data=data.yaml\
    workers=2 imgsz=640 batch=64 device=0 epochs=10

                 Class     Images  Instances      Box(P          R      mAP50  m        
                   all         28        401       0.92      0.814      0.824       0.53
                  Ball         28         23      0.922      0.515      0.512       0.28
                Player         28        346      0.905      0.988      0.992      0.686
                   Ref         28         32      0.933      0.938      0.968      0.624

### video streaming

In [6]:
import cv2
from ultralytics import YOLO
from ultralytics.utils.plotting import Annotator  # ultralytics.yolo.utils.plotting is deprecated

# Load YOLOv8 model
model = YOLO('new_best.pt')

video_path = "videoplayback.mp4"  # Specify the path to your video file
# Open video stream
cap = cv2.VideoCapture(video_path)

while True:
    # Capture frame-by-frame
    ret, frame = cap.read()
    if not ret:
        break

    # Perform inference
    results = model(frame)

    # Process detections
    for r in results:

        annotator = Annotator(frame)

        boxes = r.boxes
        for box in boxes:

            b = box.xyxy[0]  # get box coordinates in (left, top, right, bottom) format
            c = box.cls
            annotator.box_label(b, model.names[int(c)])

    frame = annotator.result()

    # Display frame
    cv2.imshow("Object Detection", frame)

    # Break loop if 'q' is pressed
    if cv2.waitKey(1) & 0xFF == ord('q'):
        break

# Release video capture and close windows
cap.release()
cv2.destroyAllWindows()



0: 384x640 1 Player, 125.9ms
Speed: 0.0ms preprocess, 125.9ms inference, 2529.1ms postprocess per image at shape (1, 3, 384, 640)

0: 384x640 1 Player, 15.7ms
Speed: 15.7ms preprocess, 15.7ms inference, 46.9ms postprocess per image at shape (1, 3, 384, 640)

0: 384x640 1 Player, 31.7ms
Speed: 0.0ms preprocess, 31.7ms inference, 15.6ms postprocess per image at shape (1, 3, 384, 640)

0: 384x640 1 Player, 15.6ms
Speed: 15.6ms preprocess, 15.6ms inference, 14.0ms postprocess per image at shape (1, 3, 384, 640)

0: 384x640 1 Player, 31.3ms
Speed: 0.0ms preprocess, 31.3ms inference, 0.0ms postprocess per image at shape (1, 3, 384, 640)

0: 384x640 1 Player, 15.6ms
Speed: 0.0ms preprocess, 15.6ms inference, 15.6ms postprocess per image at shape (1, 3, 384, 640)

0: 384x640 1 Player, 11.5ms
Speed: 4.5ms preprocess, 11.5ms inference, 15.7ms postprocess per image at shape (1, 3, 384, 640)

0: 384x640 1 Player, 15.6ms
Speed: 0.0ms preprocess, 15.6ms inference, 15.6ms postprocess per image at sh

### model optimization

In [9]:

cap.release()
cv2.destroyAllWindows()

quantization

In [ ]:
import torch
from torch.quantization import quantize_dynamic

# Load the YOLO model
model = YOLO('runs/detect/yolov8_Scoccer_Players_tuned/weights/best.pt')

# Apply post-training dynamic quantization
quantized_model = quantize_dynamic(model, dtype=torch.qint8)

# Save the quantized model
torch.save(quantized_model.state_dict(), 'quantized_yolo_model.pt')


Ultralytics YOLOv8.2.15 🚀 Python-3.11.9 torch-2.3.0+cu121 CUDA:0 (NVIDIA GeForce RTX 4070, 12282MiB)
engine/trainer: task=detect, mode=train, model=runs/detect/yolov8_Scoccer_Players_tuned/weights/best.pt, data=data.yaml, epochs=100, time=None, patience=100, batch=16, imgsz=640, save=True, save_period=-1, cache=False, device=None, workers=8, project=None, name=train2, exist_ok=False, pretrained=True, optimizer=auto, verbose=True, seed=0, deterministic=True, single_cls=False, rect=False, cos_lr=False, close_mosaic=10, resume=False, amp=True, fraction=1.0, profile=False, freeze=None, multi_scale=False, overlap_mask=True, mask_ratio=4, dropout=0.0, val=True, split=val, save_json=False, save_hybrid=False, conf=None, iou=0.7, max_det=300, half=False, dnn=False, plots=True, source=None, vid_stride=1, stream_buffer=False, visualize=False, augment=False, agnostic_nms=False, classes=None, retina_masks=False, embed=None, show=False, save_frames=False, save_txt=False, save_conf=False, save_crop=F

train: Scanning /home/mo/projects/yolo-project/datasets/Soccer-Players-16/train/labels.cache... 103 images, 1 backgrounds, 0 corrupt: 100%|██████████| 103/103 [00:00<?, ?it/s]

WARNING ⚠️ Box and segment counts should be equal, but got len(segments) = 14, len(boxes) = 1342. To resolve this only boxes will be used and all segments will be removed. To avoid this please supply either a detect or segment dataset, not a detect-segment mixed dataset.



val: Scanning /home/mo/projects/yolo-project/datasets/Soccer-Players-16/valid/labels.cache... 28 images, 1 backgrounds, 0 corrupt: 100%|██████████| 28/28 [00:00<?, ?it/s]


Plotting labels to runs/detect/train2/labels.jpg... 
optimizer: 'optimizer=auto' found, ignoring 'lr0=0.01' and 'momentum=0.937' and determining best 'optimizer', 'lr0' and 'momentum' automatically... 
optimizer: AdamW(lr=0.001429, momentum=0.9) with parameter groups 57 weight(decay=0.0), 64 weight(decay=0.0005), 63 bias(decay=0.0)
Image sizes 640 train, 640 val
Using 8 dataloader workers
Logging results to runs/detect/train2
Starting training for 100 epochs...

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      1/100      2.29G      0.952     0.5395     0.8667        198        640: 100%|██████████| 7/7 [00:01<00:00,  3.88it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 1/1 [00:00<00:00,  6.47it/s]

                   all         28        401      0.913      0.798      0.822       0.53



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      2/100      2.42G     0.9503     0.5292     0.8557        217        640: 100%|██████████| 7/7 [00:00<00:00, 10.04it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 1/1 [00:00<00:00, 10.29it/s]

                   all         28        401      0.898      0.801       0.81      0.494



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      3/100      2.43G      0.928     0.5379     0.8711        108        640: 100%|██████████| 7/7 [00:00<00:00, 10.89it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 1/1 [00:00<00:00, 10.92it/s]

                   all         28        401      0.925      0.795      0.808      0.507



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      4/100      2.54G     0.9326     0.5286      0.873        244        640: 100%|██████████| 7/7 [00:00<00:00, 12.22it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 1/1 [00:00<00:00,  8.60it/s]

                   all         28        401      0.928      0.799      0.795      0.511



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      5/100      2.43G     0.9339     0.5216     0.8593        113        640: 100%|██████████| 7/7 [00:00<00:00,  8.66it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 1/1 [00:00<00:00,  8.48it/s]

                   all         28        401      0.939      0.788      0.816      0.518



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      6/100      2.28G     0.9378     0.5459     0.8739        109        640: 100%|██████████| 7/7 [00:00<00:00, 11.48it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 1/1 [00:00<00:00,  9.41it/s]

                   all         28        401      0.911      0.783      0.799      0.517



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      7/100      2.35G     0.9651     0.5531     0.8899        105        640: 100%|██████████| 7/7 [00:00<00:00, 11.85it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 1/1 [00:00<00:00,  9.11it/s]

                   all         28        401      0.955      0.773      0.802      0.464



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      8/100      2.58G     0.9793     0.5628     0.8766        212        640: 100%|██████████| 7/7 [00:00<00:00, 11.44it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 1/1 [00:00<00:00, 10.74it/s]

                   all         28        401      0.873      0.793      0.803      0.476



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      9/100      2.54G     0.9745     0.5599     0.8717        188        640: 100%|██████████| 7/7 [00:00<00:00, 10.68it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 1/1 [00:00<00:00, 10.15it/s]

                   all         28        401      0.881      0.784      0.812      0.509



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     10/100       2.3G     0.9523     0.5396      0.883        221        640: 100%|██████████| 7/7 [00:00<00:00, 11.12it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 1/1 [00:00<00:00, 10.73it/s]

                   all         28        401      0.854      0.777      0.805      0.503



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     11/100      2.31G     0.9549     0.5417     0.8685        207        640: 100%|██████████| 7/7 [00:00<00:00, 11.93it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 1/1 [00:00<00:00, 11.44it/s]

                   all         28        401      0.889      0.797      0.808      0.509



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     12/100      2.28G     0.9666     0.5597      0.882         93        640: 100%|██████████| 7/7 [00:00<00:00, 11.24it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 1/1 [00:00<00:00,  9.36it/s]

                   all         28        401      0.938      0.756      0.806      0.495



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     13/100       2.3G     0.9771     0.5464     0.8796        163        640: 100%|██████████| 7/7 [00:00<00:00, 12.76it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 1/1 [00:00<00:00, 11.85it/s]

                   all         28        401      0.932      0.728      0.801      0.455



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     14/100      2.26G     0.9919     0.5536     0.8871        120        640: 100%|██████████| 7/7 [00:00<00:00, 12.88it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 1/1 [00:00<00:00,  9.39it/s]

                   all         28        401      0.917      0.761      0.787      0.489



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     15/100      2.29G      1.039     0.5551     0.8919        101        640: 100%|██████████| 7/7 [00:00<00:00, 12.91it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 1/1 [00:00<00:00,  8.94it/s]

                   all         28        401      0.925      0.765      0.789      0.499



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     16/100      2.26G     0.9877     0.5525     0.8781        143        640: 100%|██████████| 7/7 [00:00<00:00, 12.64it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 1/1 [00:00<00:00, 11.33it/s]

                   all         28        401      0.919      0.764      0.791      0.473



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     17/100      2.28G       0.97     0.5299     0.8689        144        640: 100%|██████████| 7/7 [00:00<00:00, 13.41it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 1/1 [00:00<00:00,  9.09it/s]

                   all         28        401      0.913      0.757       0.79      0.495



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     18/100      2.43G     0.9571      0.539     0.8727        203        640: 100%|██████████| 7/7 [00:00<00:00, 12.11it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 1/1 [00:00<00:00, 10.37it/s]

                   all         28        401      0.967      0.761       0.82      0.504



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     19/100      2.51G      0.999     0.5479     0.8963        172        640: 100%|██████████| 7/7 [00:00<00:00, 11.71it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 1/1 [00:00<00:00, 11.22it/s]

                   all         28        401      0.894       0.78      0.806      0.501



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     20/100       2.3G     0.9636      0.526     0.8827        122        640: 100%|██████████| 7/7 [00:00<00:00,  9.98it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 1/1 [00:00<00:00,  6.14it/s]

                   all         28        401      0.921      0.755      0.793       0.49



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     21/100      2.38G     0.9439     0.5383     0.8659        194        640: 100%|██████████| 7/7 [00:00<00:00, 11.27it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 1/1 [00:00<00:00,  6.72it/s]

                   all         28        401      0.891      0.772      0.817      0.501



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     22/100      2.42G     0.9612     0.5383     0.8864        112        640: 100%|██████████| 7/7 [00:00<00:00, 12.73it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 1/1 [00:00<00:00, 10.02it/s]

                   all         28        401        0.9      0.768      0.802      0.497



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     23/100      2.27G     0.9275     0.5239     0.8649        154        640: 100%|██████████| 7/7 [00:00<00:00, 12.62it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 1/1 [00:00<00:00, 12.47it/s]

                   all         28        401      0.919      0.751      0.804      0.493



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     24/100      2.53G     0.9314     0.5179     0.8703        133        640: 100%|██████████| 7/7 [00:00<00:00, 11.93it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 1/1 [00:00<00:00, 12.64it/s]

                   all         28        401      0.922      0.771      0.804      0.507



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     25/100       2.3G     0.9208     0.5188     0.8644        121        640: 100%|██████████| 7/7 [00:00<00:00, 13.28it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 1/1 [00:00<00:00, 11.65it/s]

                   all         28        401      0.849      0.777      0.785      0.495



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     26/100      2.42G     0.9194     0.5174     0.8658        244        640: 100%|██████████| 7/7 [00:00<00:00, 12.38it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 1/1 [00:00<00:00, 11.62it/s]

                   all         28        401      0.847      0.781      0.789        0.5



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     27/100      2.25G     0.9086     0.5117     0.8714        177        640: 100%|██████████| 7/7 [00:00<00:00, 12.37it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 1/1 [00:00<00:00,  8.88it/s]

                   all         28        401      0.885      0.738      0.779      0.487



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     28/100      2.26G     0.9289     0.5186      0.881        147        640: 100%|██████████| 7/7 [00:00<00:00, 11.67it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 1/1 [00:00<00:00, 11.95it/s]

                   all         28        401       0.87      0.756      0.785      0.492



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     29/100      2.45G      0.939     0.5102     0.8704        152        640: 100%|██████████| 7/7 [00:00<00:00, 12.98it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 1/1 [00:00<00:00,  9.35it/s]

                   all         28        401      0.887      0.758       0.78      0.492



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     30/100      2.42G     0.8901     0.5001     0.8697         95        640: 100%|██████████| 7/7 [00:00<00:00, 12.48it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 1/1 [00:00<00:00,  9.30it/s]

                   all         28        401      0.935       0.76      0.788      0.485



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     31/100      2.43G     0.9512     0.5178     0.8652        222        640: 100%|██████████| 7/7 [00:00<00:00,  9.57it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 1/1 [00:00<00:00,  8.07it/s]

                   all         28        401      0.901      0.758       0.79      0.468



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     32/100      2.29G     0.8884     0.4925     0.8688        128        640: 100%|██████████| 7/7 [00:00<00:00, 12.90it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 1/1 [00:00<00:00, 11.04it/s]

                   all         28        401      0.923      0.794      0.814      0.493



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     33/100      2.37G      0.885     0.4891     0.8654        140        640: 100%|██████████| 7/7 [00:00<00:00, 12.96it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 1/1 [00:00<00:00, 12.31it/s]

                   all         28        401      0.898      0.761      0.775      0.484



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     34/100      2.37G     0.8996     0.5091     0.8684        169        640: 100%|██████████| 7/7 [00:00<00:00, 13.31it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 1/1 [00:00<00:00, 10.49it/s]

                   all         28        401      0.901      0.792       0.82      0.512



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     35/100      2.28G     0.8973     0.4976     0.8656        134        640: 100%|██████████| 7/7 [00:00<00:00, 12.60it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 1/1 [00:00<00:00,  9.75it/s]

                   all         28        401      0.942      0.764      0.811      0.496



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     36/100      2.44G     0.8918     0.4868     0.8591        150        640: 100%|██████████| 7/7 [00:00<00:00, 11.76it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 1/1 [00:00<00:00, 11.59it/s]

                   all         28        401      0.919      0.768      0.805      0.498



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     37/100      2.52G     0.8921     0.4919     0.8599        182        640: 100%|██████████| 7/7 [00:00<00:00, 12.52it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 1/1 [00:00<00:00,  9.86it/s]

                   all         28        401      0.923      0.776      0.807      0.507



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     38/100      2.52G      0.883     0.4847     0.8604        192        640: 100%|██████████| 7/7 [00:00<00:00, 12.52it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 1/1 [00:00<00:00, 10.20it/s]

                   all         28        401      0.878      0.797      0.805        0.5



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     39/100      2.45G     0.8977     0.4923     0.8562        172        640: 100%|██████████| 7/7 [00:00<00:00, 11.36it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 1/1 [00:00<00:00, 10.92it/s]

                   all         28        401      0.863      0.787      0.786      0.479



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     40/100      2.45G     0.8711     0.4823     0.8528        198        640: 100%|██████████| 7/7 [00:00<00:00, 11.48it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 1/1 [00:00<00:00, 12.35it/s]

                   all         28        401      0.865      0.791      0.808      0.491



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     41/100      2.29G     0.8621     0.4787      0.856        185        640: 100%|██████████| 7/7 [00:00<00:00, 13.18it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 1/1 [00:00<00:00,  8.49it/s]

                   all         28        401      0.861      0.795      0.808       0.51



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     42/100       2.5G     0.8765     0.4842     0.8548        133        640: 100%|██████████| 7/7 [00:00<00:00, 12.90it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 1/1 [00:00<00:00, 11.60it/s]

                   all         28        401      0.872      0.827      0.838      0.523



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     43/100      2.25G     0.8751      0.486     0.8543        229        640: 100%|██████████| 7/7 [00:00<00:00, 12.47it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 1/1 [00:00<00:00, 10.57it/s]

                   all         28        401      0.895      0.776      0.813      0.512



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     44/100      2.27G     0.8635     0.4806     0.8559        165        640: 100%|██████████| 7/7 [00:00<00:00, 12.34it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 1/1 [00:00<00:00,  9.26it/s]

                   all         28        401      0.866        0.8      0.817       0.52



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     45/100       2.3G      0.877     0.4796      0.863        115        640: 100%|██████████| 7/7 [00:00<00:00, 12.31it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 1/1 [00:00<00:00, 11.79it/s]

                   all         28        401       0.84      0.792      0.811       0.51



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     46/100      2.26G     0.8786      0.479     0.8574        130        640: 100%|██████████| 7/7 [00:00<00:00,  8.82it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 1/1 [00:00<00:00, 10.89it/s]

                   all         28        401      0.901       0.77      0.817      0.513



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     47/100      2.35G     0.8811     0.4802     0.8584        252        640: 100%|██████████| 7/7 [00:00<00:00, 11.73it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 1/1 [00:00<00:00, 11.29it/s]

                   all         28        401      0.841      0.813      0.815      0.521



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     48/100      2.27G     0.8593     0.4779     0.8402        250        640: 100%|██████████| 7/7 [00:00<00:00,  9.91it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 1/1 [00:00<00:00, 11.24it/s]

                   all         28        401      0.845      0.818      0.815      0.507



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     49/100      2.26G     0.8337     0.4632      0.858        134        640: 100%|██████████| 7/7 [00:00<00:00, 12.20it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 1/1 [00:00<00:00, 12.66it/s]

                   all         28        401      0.863      0.812      0.816      0.504



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     50/100      2.27G     0.8374     0.4656     0.8499        134        640: 100%|██████████| 7/7 [00:00<00:00, 12.07it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 1/1 [00:00<00:00, 12.74it/s]

                   all         28        401      0.923      0.788      0.808      0.513



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     51/100      2.25G     0.8348     0.4632     0.8455        174        640: 100%|██████████| 7/7 [00:00<00:00, 11.74it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 1/1 [00:00<00:00, 13.00it/s]

                   all         28        401      0.888       0.77       0.79      0.501



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     52/100      2.41G     0.8413     0.4666     0.8535        188        640: 100%|██████████| 7/7 [00:00<00:00,  9.24it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 1/1 [00:00<00:00, 12.24it/s]

                   all         28        401      0.925      0.785       0.81      0.515



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     53/100      2.27G      0.822     0.4646     0.8754        136        640: 100%|██████████| 7/7 [00:00<00:00, 13.12it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 1/1 [00:00<00:00, 10.82it/s]

                   all         28        401      0.922      0.783      0.811       0.52



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     54/100      2.39G     0.8005     0.4493     0.8493        185        640: 100%|██████████| 7/7 [00:00<00:00,  8.22it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 1/1 [00:00<00:00,  9.53it/s]

                   all         28        401      0.937      0.783      0.818      0.522



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     55/100      2.25G     0.8105     0.4533     0.8495        149        640: 100%|██████████| 7/7 [00:00<00:00, 11.98it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 1/1 [00:00<00:00, 10.11it/s]

                   all         28        401      0.931      0.783      0.823      0.535



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     56/100      2.46G     0.8197     0.4589     0.8502        176        640: 100%|██████████| 7/7 [00:00<00:00, 12.89it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 1/1 [00:00<00:00, 13.13it/s]

                   all         28        401       0.93      0.776      0.819      0.523



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     57/100      2.36G     0.8233     0.4609     0.8617        172        640: 100%|██████████| 7/7 [00:00<00:00, 12.81it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 1/1 [00:00<00:00, 10.80it/s]

                   all         28        401      0.937      0.775      0.818      0.524



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     58/100      2.29G     0.8371      0.455     0.8532        144        640: 100%|██████████| 7/7 [00:00<00:00, 13.08it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 1/1 [00:00<00:00, 12.90it/s]

                   all         28        401      0.939      0.775      0.817      0.513



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     59/100      2.44G     0.8389     0.4615     0.8543        110        640: 100%|██████████| 7/7 [00:00<00:00, 12.34it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 1/1 [00:00<00:00, 10.53it/s]

                   all         28        401      0.933      0.775      0.815       0.52



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     60/100      2.29G     0.8011     0.4474     0.8446        258        640: 100%|██████████| 7/7 [00:00<00:00,  9.49it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 1/1 [00:00<00:00,  5.98it/s]

                   all         28        401      0.933      0.776       0.83      0.529



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     61/100       2.3G     0.7926     0.4424     0.8332        205        640: 100%|██████████| 7/7 [00:00<00:00, 13.47it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 1/1 [00:00<00:00, 13.09it/s]

                   all         28        401      0.937      0.776      0.834      0.533



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     62/100      2.43G     0.8094     0.4472     0.8461        155        640: 100%|██████████| 7/7 [00:00<00:00, 12.48it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 1/1 [00:00<00:00,  9.72it/s]

                   all         28        401      0.981      0.774      0.832      0.532



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     63/100      2.44G     0.7839     0.4406     0.8529        171        640: 100%|██████████| 7/7 [00:00<00:00, 12.34it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 1/1 [00:00<00:00,  7.04it/s]


                   all         28        401       0.88      0.786      0.823       0.54

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     64/100      2.47G     0.8296     0.4489     0.8493        162        640: 100%|██████████| 7/7 [00:00<00:00, 12.12it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 1/1 [00:00<00:00, 10.48it/s]

                   all         28        401      0.847        0.8      0.821      0.529



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     65/100      2.27G     0.7816     0.4386      0.851        154        640: 100%|██████████| 7/7 [00:00<00:00, 13.31it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 1/1 [00:00<00:00, 12.17it/s]

                   all         28        401      0.899      0.784      0.819      0.521



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     66/100      2.26G     0.7773      0.437     0.8505        123        640: 100%|██████████| 7/7 [00:00<00:00, 12.54it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 1/1 [00:00<00:00, 11.96it/s]

                   all         28        401       0.88      0.816       0.82      0.519



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     67/100      2.25G     0.8064     0.4421     0.8444        225        640: 100%|██████████| 7/7 [00:00<00:00, 12.77it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 1/1 [00:00<00:00, 12.16it/s]

                   all         28        401      0.902      0.795      0.815      0.524



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     68/100      2.42G     0.7888      0.441     0.8453        173        640: 100%|██████████| 7/7 [00:00<00:00, 12.26it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 1/1 [00:00<00:00, 11.30it/s]

                   all         28        401      0.896      0.799       0.82      0.523



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     69/100      2.56G     0.7855     0.4404     0.8479        133        640: 100%|██████████| 7/7 [00:00<00:00, 12.19it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 1/1 [00:00<00:00, 11.81it/s]

                   all         28        401      0.934      0.783       0.82      0.529



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     70/100      2.37G     0.7746     0.4331     0.8455        201        640: 100%|██████████| 7/7 [00:00<00:00, 12.26it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 1/1 [00:00<00:00, 12.18it/s]

                   all         28        401      0.948      0.768       0.82      0.538



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     71/100      2.28G     0.7794     0.4335      0.848        128        640: 100%|██████████| 7/7 [00:00<00:00, 11.43it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 1/1 [00:00<00:00,  6.25it/s]

                   all         28        401      0.869      0.816      0.822      0.526



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     72/100      2.28G     0.7449     0.4228      0.837        142        640: 100%|██████████| 7/7 [00:00<00:00, 12.41it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 1/1 [00:00<00:00, 11.29it/s]

                   all         28        401      0.868      0.815      0.822      0.526



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     73/100      2.36G     0.7651     0.4302      0.846        214        640: 100%|██████████| 7/7 [00:00<00:00, 11.73it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 1/1 [00:00<00:00, 11.69it/s]

                   all         28        401      0.903      0.785      0.823      0.521



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     74/100       2.5G     0.7353     0.4196     0.8422        127        640: 100%|██████████| 7/7 [00:00<00:00, 12.56it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 1/1 [00:00<00:00, 10.25it/s]

                   all         28        401      0.979      0.758       0.82      0.524



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     75/100      2.41G     0.7817     0.4296     0.8497        116        640: 100%|██████████| 7/7 [00:00<00:00, 12.29it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 1/1 [00:00<00:00, 12.36it/s]

                   all         28        401      0.866        0.8      0.821      0.528



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     76/100      2.46G     0.7461     0.4138     0.8284        202        640: 100%|██████████| 7/7 [00:00<00:00, 11.82it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 1/1 [00:00<00:00, 10.77it/s]

                   all         28        401      0.904      0.816      0.824      0.532



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     77/100      2.31G     0.7287     0.4147     0.8393        107        640: 100%|██████████| 7/7 [00:00<00:00, 11.47it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 1/1 [00:00<00:00,  7.56it/s]

                   all         28        401      0.936      0.809      0.826      0.533



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     78/100       2.4G       0.73     0.4195     0.8356        148        640: 100%|██████████| 7/7 [00:00<00:00, 12.68it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 1/1 [00:00<00:00, 10.87it/s]

                   all         28        401      0.929      0.816      0.825      0.539



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     79/100      2.39G     0.7359     0.4156     0.8352        192        640: 100%|██████████| 7/7 [00:00<00:00, 11.29it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 1/1 [00:00<00:00,  5.60it/s]


                   all         28        401      0.909      0.801      0.822      0.526

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     80/100      2.31G     0.7462     0.4246     0.8384        133        640: 100%|██████████| 7/7 [00:00<00:00, 12.04it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 1/1 [00:00<00:00, 12.35it/s]

                   all         28        401      0.907        0.8      0.813      0.523



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     81/100      2.28G     0.7359     0.4248     0.8431        142        640: 100%|██████████| 7/7 [00:00<00:00, 13.35it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 1/1 [00:00<00:00, 11.66it/s]

                   all         28        401      0.902      0.809      0.817       0.52



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     82/100      2.25G     0.7579     0.4208     0.8511        182        640: 100%|██████████| 7/7 [00:00<00:00, 12.37it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 1/1 [00:00<00:00, 12.39it/s]

                   all         28        401      0.887      0.791      0.805      0.523



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     83/100      2.27G      0.763     0.4276     0.8407        212        640: 100%|██████████| 7/7 [00:00<00:00, 12.14it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 1/1 [00:00<00:00, 11.04it/s]

                   all         28        401      0.897      0.818      0.819      0.526



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     84/100      2.44G       0.74     0.4221     0.8411        154        640: 100%|██████████| 7/7 [00:00<00:00, 12.29it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 1/1 [00:00<00:00, 12.43it/s]

                   all         28        401      0.951      0.798      0.819      0.521



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     85/100      2.45G     0.7616     0.4271     0.8399        204        640: 100%|██████████| 7/7 [00:00<00:00, 12.63it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 1/1 [00:00<00:00, 10.25it/s]

                   all         28        401       0.95        0.8      0.821      0.525



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     86/100      2.27G     0.7026     0.4082      0.832        139        640: 100%|██████████| 7/7 [00:00<00:00,  9.38it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 1/1 [00:00<00:00, 11.32it/s]

                   all         28        401      0.938        0.8      0.809      0.521



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     87/100      2.28G      0.736     0.4155     0.8303        113        640: 100%|██████████| 7/7 [00:00<00:00, 11.88it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 1/1 [00:00<00:00,  7.95it/s]

                   all         28        401      0.941        0.8       0.82      0.525



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     88/100       2.3G     0.7057       0.41     0.8363        138        640: 100%|██████████| 7/7 [00:00<00:00, 10.67it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 1/1 [00:00<00:00, 10.54it/s]

                   all         28        401      0.938      0.802      0.819      0.523



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     89/100      2.43G      0.707     0.4117     0.8372        223        640: 100%|██████████| 7/7 [00:00<00:00, 12.53it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 1/1 [00:00<00:00,  9.24it/s]

                   all         28        401      0.912      0.816      0.819       0.52



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     90/100      2.26G     0.7175     0.4087     0.8372        157        640: 100%|██████████| 7/7 [00:00<00:00, 13.39it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 1/1 [00:00<00:00, 12.60it/s]


                   all         28        401       0.91      0.816      0.824      0.522
Closing dataloader mosaic

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     91/100      2.15G     0.7143     0.4207     0.8324         87        640: 100%|██████████| 7/7 [00:03<00:00,  2.02it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 1/1 [00:00<00:00, 11.25it/s]

                   all         28        401      0.908      0.815      0.825      0.516



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     92/100      2.16G     0.6801     0.4115     0.8325         76        640: 100%|██████████| 7/7 [00:00<00:00, 13.46it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 1/1 [00:00<00:00, 10.84it/s]

                   all         28        401      0.924      0.802      0.821      0.517



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     93/100      2.16G     0.6955     0.4057     0.8337         83        640: 100%|██████████| 7/7 [00:00<00:00, 13.31it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 1/1 [00:00<00:00, 10.46it/s]

                   all         28        401       0.92      0.802      0.821      0.514



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     94/100      2.16G     0.6766     0.3999     0.8325         74        640: 100%|██████████| 7/7 [00:00<00:00, 13.65it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 1/1 [00:00<00:00, 11.16it/s]

                   all         28        401       0.91      0.802      0.821      0.524



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     95/100      2.15G     0.7086      0.413      0.833         81        640: 100%|██████████| 7/7 [00:00<00:00, 13.17it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 1/1 [00:00<00:00, 11.00it/s]

                   all         28        401      0.906      0.801       0.82      0.519



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     96/100      2.16G     0.6459     0.3885     0.8367         35        640: 100%|██████████| 7/7 [00:00<00:00, 13.09it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 1/1 [00:00<00:00, 10.85it/s]

                   all         28        401      0.885      0.801      0.819      0.515



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     97/100      2.16G     0.6731     0.4009     0.8234        111        640: 100%|██████████| 7/7 [00:00<00:00, 14.13it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 1/1 [00:00<00:00, 11.16it/s]

                   all         28        401      0.883      0.802      0.817      0.511



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     98/100      2.16G     0.6652     0.3948     0.8343         78        640: 100%|██████████| 7/7 [00:00<00:00, 13.02it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 1/1 [00:00<00:00, 12.04it/s]

                   all         28        401      0.885      0.802      0.817      0.517



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     99/100      2.15G     0.6431     0.3845     0.8286         83        640: 100%|██████████| 7/7 [00:00<00:00, 12.87it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 1/1 [00:00<00:00, 11.18it/s]

                   all         28        401      0.887      0.802      0.815      0.516



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


    100/100      2.16G     0.6509     0.3903     0.8265         88        640: 100%|██████████| 7/7 [00:00<00:00, 12.19it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 1/1 [00:00<00:00,  8.83it/s]

                   all         28        401      0.886      0.802      0.813      0.518



100 epochs completed in 0.029 hours.
Optimizer stripped from runs/detect/train2/weights/last.pt, 6.2MB
Optimizer stripped from runs/detect/train2/weights/best.pt, 6.2MB

Validating runs/detect/train2/weights/best.pt...
Ultralytics YOLOv8.2.15 🚀 Python-3.11.9 torch-2.3.0+cu121 CUDA:0 (NVIDIA GeForce RTX 4070, 12282MiB)
Model summary (fused): 168 layers, 3006233 parameters, 0 gradients, 8.1 GFLOPs


                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 1/1 [00:00<00:00,  6.54it/s]


                   all         28        401      0.878      0.786      0.823      0.541
                  Ball         28         23       0.77      0.435      0.517      0.286
                Player         28        346      0.921      0.986      0.993      0.709
                   Ref         28         32      0.942      0.938      0.958      0.626
Speed: 0.1ms preprocess, 0.5ms inference, 0.0ms loss, 0.5ms postprocess per image
Results saved to runs/detect/train2


### evaluation of quantization
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95):
                   all         28        401      0.878      0.786      0.823      0.541
                  Ball         28         23       0.77      0.435      0.517      0.286
                Player         28        346      0.921      0.986      0.993      0.709
                   Ref         28         32      0.942      0.938      0.958      0.626

pruning

In [ ]:
import torch.nn.utils.prune as prune

model = YOLO('runs/detect/yolov8_Scoccer_Players_tuned/weights/best.pt')

for name, layer in model.named_modules():
    if 'conv' in name:
        prune.l1_unstructured(layer, name='weight', amount=.2)


torch.save(model.state_dict(), 'pruned_yolo_model.pt')